lms/djangoapps/instructor/tasks.py

```diff
@@ -0,0 +1,96 @@
+""" Celery Tasks for the Instructor App """
+
+import logging
+
+from celery import shared_task
+from celery_utils.logged_task import LoggedTask
+from django.core.exceptions import ObjectDoesNotExist
+from opaque_keys import InvalidKeyError
+from opaque_keys.edx.keys import UsageKey
+from xblock.completable import XBlockCompletionMode
+
+from common.djangoapps.student.models import get_user_by_username_or_email
+from lms.djangoapps.courseware.model_data import FieldDataCache
+from lms.djangoapps.courseware.module_render import get_module_for_descriptor
+from openedx.core.lib.request_utils import get_request_or_stub
+from xmodule.modulestore.django import modulestore
+from xmodule.modulestore.exceptions import ItemNotFoundError
+
+log = logging.getLogger(__name__)
+
+
+@shared_task(base=LoggedTask, ignore_result=True)
+def complete_student_attempt_task(user_identifier, content_id):
+    """
+    Submits all completable xblocks inside of the content_id block to the
+    Completion Service to mark them as complete. One use case of this function is
+    for special exams (timed/proctored) where regardless of submission status on
+    individual problems, we want to mark the entire exam as complete when the exam
+    is finished.
+
+    params:
+        user_identifier (String): username or email of a user
+        content_id (String): the block key for a piece of content
+    """
+    err_msg_prefix = (
+        'Error occurred while attempting to complete student attempt for user '
+        f'{user_identifier} for content_id {content_id}. '
+    )
+    err_msg = None
+    try:
+        user = get_user_by_username_or_email(user_identifier)
+        block_key = UsageKey.from_string(content_id)
+        root_descriptor = modulestore().get_item(block_key)
+    except ObjectDoesNotExist:
+        err_msg = err_msg_prefix + 'User does not exist!'
+    except InvalidKeyError:
+        err_msg = err_msg_prefix + 'Invalid content_id!'
+    except ItemNotFoundError:
+        err_msg = err_msg_prefix + 'Block not found in the modulestore!'
+    if err_msg:
+        log.error(err_msg)
+        return
+
+    # This logic has been copied over from openedx/core/djangoapps/schedules/content_highlights.py
+    # in the _get_course_module function.
+    # I'm not sure if it's a pattern that should be copied so likely if it can be avoided, do that.
```

Comment:
This didn't scan easily for me. Maybe something like `I'm not sure if this is an anti-pattern or not, so if you can avoid re-copying this, please do`?